fn_cve_search/fn_cve_search/components/function_cve.py

```diff
@@ -0,0 +1,269 @@
+
+            _MAX_RESULTS_RETURN = int(self.options.get('max_results_display'))
+
+            log = logging.getLogger(__name__)
+
+            log.info("Maximum search result to be returned : {}".format(_MAX_RESULTS_RETURN))
+            log.info("cve_search_data: %s type : %s", cve_search_data,type(cve_search_data))
+            log.info("cve_search_criteria: %s type : %s", cve_search_criteria,type(cve_search_criteria))
+            log.info("cve_id: %s type : %s", cve_id,type(cve_id))
+            log.info("cve_vendor: %s type : %s", cve_vendor,type(cve_vendor))
+            log.info("cve_product: %s type : %s", cve_product,type(cve_product))
+            log.info("cve_published_date_from: %s type : %s", cve_published_date_from,type(cve_published_date_from))
+            log.info("cve_published_date_to: %s type : %s", cve_published_date_to,type(cve_published_date_to))
+
+            yield StatusMessage("starting...")
+            _browse_data = None   # A variable to store the returned JSON Data
+            if cve_search_criteria.lower().find('browse') != -1:
+                if cve_vendor is not None and cve_product is not None:
+                    _browse_data = self._search_cve_api(vendor_name=cve_vendor, product=cve_product)
+                else:
+                    _browse_data = self._browse_cve_api(vendor_name=cve_vendor)
+            elif cve_search_criteria.lower().find('search') != -1:
+                if cve_vendor is None and cve_product is None:
+                    _browse_data = self._browse_cve_api(vendor_name=cve_vendor)
+                else:
+                    _browse_data = self._search_cve_api(vendor_name=cve_vendor, product=cve_product)
+            elif cve_search_criteria.lower().find('specific') != -1:
+                _browse_data = self._get_specific_cve_data(cve_id=cve_id)
+            elif cve_search_criteria.lower().find('last') != -1:
+                _browse_data = self._last_30_cves()
+            else:
+                _browse_data = self._cve_db_information()
+
+            # Filtering the date to send back to resilient
+            _result_data_dict = dict()
+            _result_data_dict['content'] = []
+
+            # Type Of the rest api call like browse,search,specific cve,last, db info
+            _api_call_type = _browse_data['api_call']
+            # Rest Api Response Data
+            _browse_data_content = _browse_data['content']
+            if _api_call_type == 'browse':
+                _result_data_dict['api_call'] = 'browse'
+                if isinstance(_browse_data_content, list):
+                    raise NotImplementedError()
+                elif isinstance(_browse_data_content, dict):
+                    for key_data, value_data in _browse_data_content.items():
+                        _result_data_dict['content'].append({key_data: value_data})
+                else:
+                    raise NotImplementedError()
+            elif _api_call_type == 'search':
```

Comment:
this is a massive function. Can you split up into smaller modular functions?

Reply:
ok